### DISCUSSION TITLE
How to manually trigger scan for new photos in the External library?

### DISCUSSION THREAD

**Discussion by SlavikCA** [immich-app/immich#6236] [Q&A]
I'm running Immich in Docker.
I'm adding photos manually from my cameras.

I think Immich configured to run scan every night. 

Is there a CLI / docker command I can run to start scan manually?

Also, looking at `UI -> Administration -> Jobs`, which job I should start to scan for new photos:
- `Library -> Refresh`? I tried and I think it scan everything now. Also, files which I deleted are still present after the `refresh`
- `Extract Metadata - Missing`?
- `Generate Thumbnails -> Missing`?

**Comment by zodcas** [comment]
I looked into using the API for these type of things, could never get it to work how I wanted so I caved and stopped using the external library. You can give it a crack though:
Scanning, using the API @ https://immich.app/docs/api/scan-library

Library refresh of deleted files, using the API @ https://immich.app/docs/api/remove-offline-files

**Reply by mojothemonkey2** [reply]
thanks, that api worked ok for me to trigger scan of external libarary. eg:
```
API_KEY=****
LIBRARY_ID=****
curl -H "x-api-key: $API_KEY" -X POST "http://127.0.0.1:2283/api/libraries/$LIBRARY_ID/scan"
```
where api key can be added in: Account Settings > Api keys (needs `library.update` permission),
and library id can be found by inspecting the External Libraries page, and watching network, then triggering a scan manually from the page. (maybe there is a better place to find it).

**Reply by polm** [reply]
I found the easiest way to get the library ID is to go to `https://example.com/api/libraries`, which will give you json with all the libraries.

**Reply by Retrospector** [reply]
Thanks for the easy example, @mojothemonkey2! 
I am syncing Photos from my Home Server to the Immich instance and was looking for a way to automatically refresh the Library after a sync was done, works a treat :rocket:

**Reply by pihomeserver** [reply]
Thank you @mojothemonkey2 
Was easy to sync a S3 bucket to my local instance and request a library refresh

**Comment by SlavikCA** [comment]
Thank you. That `remove-offline-files` API request actually did what I wanted to do.
I wonder why it's not on by default, when I do `Library -> Refresh`.

**Comment by haydonryan** [comment]
100% frustrated by the lack of main library scan. I recently migrated servers for my hosting and have had to deal with the same issue.   

Honestly I find the UI non-intuitive.  Administration is under user, so my brain thinks this as 
"User -> Adminstration" 
instead of
"Server -> Settings"

I personally find it hard here, as opposed to other self host software  like navidrome / plex where the rescan is where i expect it.

It would be much more intuitive to have a cog next to the user icon for users who have admin capability.
